### PR TITLE
Add support for authorization mode environment variable in C++

### PR DESCRIPTION
**Pull request by vekterli**
@havardpe @bjorncs please review

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]

File: vespalib/src/vespa/vespalib/net/tls/authorization_mode.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "authorization_mode.h"
+
+#include <iostream>
+
+namespace vespalib::net::tls {
+
+const char* to_string(AuthorizationMode mode) noexcept {
```
Comment (havardpe) [inline vespalib/src/vespa/vespalib/net/tls/authorization_mode.cpp]: to_string combined with string values that does not match the 'from string' behavior can be somewhat confusing. Consider using symmetric string mapping or renaming this function.

**Review by havardpe — APPROVED** [review]
👍 
minor nitpick

File: vespalib/src/vespa/vespalib/net/tls/authorization_mode.cpp (hunk 1 above)
Comment (vekterli) [reply vespalib/src/vespa/vespalib/net/tls/authorization_mode.cpp]: Renamed to `enum_name`, PTAL

**Review by havardpe — APPROVED** [review]
👍

**Pull request merged**